app/src/gui/components/notebook/record_table.tsx

```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
         },
       ];
 
+  // Fetch the current user when the component mounts
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const userId = await getCurrentUserId(project_id);
+        setCurrentUser(userId);
+      } catch (error) {
+        console.error('Error fetching user ID:', error);
+      }
+    };
+    fetchUser();
+  }, [project_id]);
+
+  useEffect(() => {
+    if (!rows || rows.length === 0 || !currentUser) {
+      return;
+    }
+
+    const totalRecords = getTotalRecordCount(rows);
+
+    const myRecords = rows.filter(
+      record =>
+        getRowType({row: record} as GridCellParams) === 'Site' &&
+        record.created_by === currentUser
+    ).length;
```

Comment:
not using helper function? Maybe could make this a helper function or add optional filter of row type

Reply:
code updated. 
:-)